### PR TITLE
Remove Regressive Smash I Cases.

### PR DESCRIPTION
**Pull request by jnoecker**
Fixes https://github.com/evllabs/JGAAP/issues/105

---
This change is [Reviewable](https://reviewable.io/reviews/evllabs/jgaap/106)

### CODE REVIEW

File: src/com/jgaap/canonicizers/SmashI.java [hunk 1]
```diff
@@ -20,15 +20,23 @@
 import com.jgaap.generics.CanonicizationException;
 import com.jgaap.generics.Canonicizer;
 
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.junit.Assert.assertTrue;
```
Comment (mvryan) [inline src/com/jgaap/canonicizers/SmashI.java]: I don't believe this import is used.

**Comment by DavidBerdik** [comment]
Nice work condensing it to two lines. I really should learn regex. The only problem I have with this is that the use of I in nonsense contractions will be detected as well. Are we okay with "I'esxdrcftvgbh" being converted to "i'esxdrcftvgbh"?

**Comment by mvryan** [comment]
What is the goal of 'Smash I'? 

Why is only one word having it's case set to lower?

**Comment by juola** [comment]
Looking at capitalized entities.   The student involved wants to look at
words surrounding named entities, and "I" messes the stats up.

Patrick

**Comment by DavidBerdik** [comment]
@jnoecker So do we not care then?

**Comment by juola** [comment]
I'd not say that we don't care, but that it's not important enough to
derail the marginal improvement in functionality that this represents.

The "Smash I" canonicizer may or may not provide an improvement in overall
performance, especially when combined with other aspects of JGAAP such as
named entity recognition.   The ability to smash I's in contractions may or
may not provide further improvements.  The side effect of smashing I's in
"nonsense contractions" may or may not provide a performance hit.  All of
these are empirical questions and we can't answer them until large-scale
tests have been done.

However, NOT writing this canonicizer is guaranteed to provide no
improvement.

Please don't allow the abstract perfect to become the enemy of the concrete
good.   If you have a better way of doing "Smash I" that does not have the
undesirable side effects, code it up and we'll run the same large-scale
tests to see if it works.    If you don't have an alternative set of code,
but you don't trust this particular canonicizer, you don't need to use it.
 The JGAAP design encourages the development of exploratory systems to see
if they work and the incremental improvement of those that do.

**Pull request merged**